packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -616,63 +658,43 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
-        // proposal has passed the challenge period.
-        optimisticOracle.requestPrice(
-            bridgeAdmin.identifier(),
-            requestTimestamp,
-            customAncillaryData,
-            IERC20(l1Token),
-            0
-        );
-
-        // Set the Optimistic oracle liveness for the price request.
-        optimisticOracle.setCustomLiveness(
-            bridgeAdmin.identifier(),
-            requestTimestamp,
-            customAncillaryData,
-            uint256(bridgeAdmin.optimisticOracleLiveness())
-        );
-
-        // Set the Optimistic oracle proposer bond for the price request.
-        uint256 proposerBond = _getProposerBond(amount);
-        optimisticOracle.setBond(bridgeAdmin.identifier(), requestTimestamp, customAncillaryData, proposerBond);
-    }
-
-    function _proposeOraclePriceRelay(
-        uint256 amount,
-        uint256 requestTimestamp,
-        bytes memory customAncillaryData
-    ) private {
-        OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
+        // Compute total proposal bond and pull from caller so that the OptimisticOracle can subsequently pull it from
+        // here.
         uint256 proposerBondPct =
             FixedPoint.Unsigned(uint256(bridgeAdmin.proposerBondPct())).div(FixedPoint.fromUnscaledUint(1)).rawValue;
         uint256 finalFee = _getStore().computeFinalFee(address(l1Token)).rawValue;
-
         uint256 totalBond =
             FixedPoint
                 .Unsigned(proposerBondPct)
                 .mul(FixedPoint.Unsigned(amount))
                 .add(FixedPoint.Unsigned(finalFee))
                 .rawValue;
-
-        // Pull the total bond from the caller so that the OptimisticOracle can subsequently pull it from here.
         l1Token.safeTransferFrom(msg.sender, address(this), totalBond);
         l1Token.safeApprove(address(optimisticOracle), totalBond);
-        optimisticOracle.proposePriceFor(
-            msg.sender,
-            address(this),
+
+        optimisticOracle.requestAndProposePriceFor(
             bridgeAdmin.identifier(),
-            requestTimestamp,
+            uint32(requestTimestamp),
             customAncillaryData,
-            1e18 // Canonical value representing "True"; i.e. the proposed relay is valid.
+            IERC20(l1Token),
+            // Set reward to 0, since we'll settle proposer reward payouts directly from this contract after a relay
+            // proposal has passed the challenge period.
+            0,
+            // Set the Optimistic oracle proposer bond for the price request.
+            _getProposerBond(amount),
+            // Set the Optimistic oracle liveness for the price request.
+            uint256(bridgeAdmin.optimisticOracleLiveness()),
+            // Caller is proposer.
+            msg.sender,
+            // Canonical value representing "True"; i.e. the proposed relay is valid.
+            1e18
```

Comment:
is it not `int(1e18)`?

Reply:
Doesn't this automatically cast it? I can be more explicit I'll change it